src/gridcoin/voting/poll.cpp

```diff
@@ -317,3 +384,34 @@ void ChoiceList::Add(std::string label)
 {
     m_choices.emplace_back(std::move(label));
 }
+
+//!
+//! \brief This allows use of range based loops through the PollType enum
+//!
+const std::vector<GRC::PollType> Poll::POLL_TYPES = {
```

Comment:
```suggestion
const GRC::PollType Poll::POLL_TYPES[] = {
```
Could avoid using heap here.

Reply:
Going to a straight array will break the use of the range based loops.

Reply:
Are you sure? This compiles under clang:
```cpp
#include <iostream>

int main() {
    int arr[] = {0, 1, 2, 3, 4};

    for (const auto& elem : arr) {
        std::cout << elem << std::endl;
    }

    return 0;
}
```

Reply:
If it's compiler-dependant, std::array would still be better than std::vector.

Reply:
That isn't the issue. In your example above you don't have a separate header.

Reply:
When you have a separate header and have
    static const GRC::PollType POLL_TYPES[];
in the header and the definition in the .cpp file, you will get a incomplete type compilation error for the range based loops in payload.h. There currently is no payload.cpp file.

Reply:
I'll use std::array. That will work.